omero/developers/Web/CreateApp.rst

```diff
@@ -6,86 +6,93 @@ to get you familiar with the Django framework. The more you know about
-use the correct port number, e.g.
-
--  `http://localhost:4080/webclient/ <http://localhost:4080/webclient/>`_
-
-When you edit and save your app, Django will automatically detect this and you
-only need to refresh your browser to see the changes.
+In order to deploy OMERO.web in a development or testing environment,
+you can use Docker as described below (recommended) or 
+follow the install instructions under :doc:`Deployment`.
 
 If you want to make changes to the OMERO.web code itself, go to
 :doc:`/developers/Web/EditingOmeroWeb`.
 
-You can place your app anywhere on your :envvar:`PYTHONPATH`,
-as long as it can be imported by OMERO.web.
+Create an app from the template example
+---------------------------------------
 
-Creating an app
----------------
+Go to `https://github.com/will-moore/omero-web-apps-examples <https://github.com/will-moore/omero-web-apps-examples>`_.
+Click 'Use this template' as `described here
+<https://help.github.com/en/articles/creating-a-repository-from-a-template>`_
+and choose a name for your new app, for example ``omero-demo-webapp``.
 
-We suggest you use `GitHub <https://github.com/>`_ (as we do) since it is much easier for us to
-help you with any problems you have if we can see your code. The steps below
-describe how to create a stand-alone git repository for your app, similar to
-`omero-webtest <https://github.com/openmicroscopy/omero-webtest>`_.
-If you do not want to use GitHub, simply ignore the steps related to GitHub.
+Go to the directory where you want your app to live and clone it:
 
-The steps below describe how to set up a new app. You should choose an
-appropriate name for your app and use it in place of <organization-appname>:
+::
 
+    $ cd /path/to/dir/
+    $ git clone https://github.com/will-moore/omero-demo-webapp
 
-Add your app to your PYTHONPATH
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+You can now use `omero-web-docker <https://github.com/ome/omero-web-docker/>`_
+to run the app. Here we add ``minimal_webapp`` to the installed apps and map the
+directory to the ``site-packages`` directory in the Docker instance so that our
+app is on the :envvar:`PYTHONPATH`, described in more detail below.
+We also choose the OMERO server we want to connect to.
+You need to edit the ``demo.openmicroscopy.org`` and ``/path/to/dir/`` in the
+following command:
 
-Your app needs to be within a directory that is on your :envvar:`PYTHONPATH`.
-We usually create a new container for a new app, and add it to the
-:envvar:`PYTHONPATH`.
 ::
 
-    $ mkdir organization-appname
-    $ cd organization-appname
-    $ export PYTHONPATH=$PYTHONPATH:/path/to/organization-appname
+    $ docker run -it -e OMEROHOST=demo.openmicroscopy.org -p 4080:4080 -e CONFIG_omero_web_apps='["minimal_webapp"]' -v /path/to/dir/omero-demo-webapp/minimal-webapp/minimal_webapp:/opt/omero/web/venv/lib/python2.7/site-packages/minimal_webapp openmicroscopy/omero-web-standalone
 
+This will run Docker interactively in the terminal.
```

Comment:
Actually not exactly. It runs the docker successfully, but the command prompt does not return ever for me to do any meaningful actions inside the docker (i.e. there is nothing interactive in this docker run).
Even if it would, I would have no clue where to start. Or should I be starting any actions inside the docker ?
Not clear from the whole doc. At the moment, it looks to me like I have to do everything via changing the cloned repo from the template. Why do I need to run "interactive" docker then ?

Reply:
I'm probably a bit wrong on the Docker terminology here. We're running it with ```-it``` which I believed meant "interactively". But googling suggests it "changes the terminal
to the terminal within the container". I just found it useful to see the output when the container runs. You don't need to do anything inside docker.

I'd like to run the docker OMERO.web in debug mode, so it's more useful for development.
Certainly, all the work happens on the repo cloned from the template.

Reply:
Then maybe state something like ``Running the docker in this way allows you to watch some useful output of the runnning of the container. The development of your app is going to happen in the repo you just cloned from the template, see the details below.``